contracts/FlowIDTableStaking.cdc

```diff
@@ -1634,14 +1657,21 @@ pub contract FlowIDTableStaking {
         return nodeIDs.keys
     }
 
-    /// Adds a node and/or a delegator to the list of node IDs who have pending token movements
-    /// or whose delegators have pending movements
-    access(contract) fun setNewMovesPending(nodeID: String, delegatorID: UInt32?) {
+    /// Adds a node and/or delegator to the moves pending list
+    /// and returns the list
+    /// If `existingList` is `nil`, this loads the current list from storage and modifies it
+    /// If `existingList` is non-`nil`, this modifies the given list instead of loading from storage
+    access(contract) fun modifyNewMovesPending(nodeID: String,
+                                               delegatorID: UInt32?,
+                                               existingList: {String: {UInt32: Bool}}?)
+                                               : {String: {UInt32: Bool}} 
```

Comment:
Passing a dictionary and and returning a dictionary is very expensive: For each call of the function, the argument for the parameter is copied, modified in the function, and then copied once more.

If the dictionary should be modified, make the parameter a reference. Further, don't load (copy and remove) the input from storage (line below), but simply borrow from storage.

Reply:
yep, I've already made this change locally. Just haven't pushed it yet